### PR TITLE
PR for Command: encrypt

### PR DESCRIPTION
**Pull request by an7e**
PR for #39 

Changes proposed in this PR:

- added command encrypt
- added encrypt handler
- added aes

### CODE REVIEW

File: src/components/core/encryptHandler.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    let blobData: Uint8Array = new Uint8Array()
+    if (task.encoding === 'String') {
+      blobData = Uint8Array.from(Buffer.from(task.blob, 'hex'))
+    }
+    if (task.encoding === 'Base58') {
+      blobData = base58.base58_to_binary(task.blob)
+    }
+
+    let encryptedData: Buffer
+    if (task.encryptionType === 'AES') {
```
Comment (alexcos20) [inline src/components/core/encryptHandler.ts:20]: where is ECIES Encryption?

File: src/components/core/encryptHandler.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      const algorithm = 'aes-256-cbc'
+      const initVector = crypto.randomBytes(16)
+      const securityKey = crypto.randomBytes(32)
```
Comment (alexcos20) [inline src/components/core/encryptHandler.ts:23]: if all encryption keys are random, who is going to be able to decrypt this?

File: src/components/core/encryptHandler.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { EncryptCommand } from '../../utils/constants'
```
Comment (mariacarmina) [inline src/components/core/encryptHandler.ts:1]: Please fix the import by importing the `.js` extension

File: src/utils/constants.ts [hunk 4]
```diff
@@ -1,9 +1,11 @@
 import { Hashes } from '../@types/blockchain'
+import { handleEncryptCommand } from '../components/core/encryptHandler'
```
Comment (mariacarmina) [inline src/utils/constants.ts:2]: Same here, I suspect that here is also a circular import

**Review by mariacarmina — COMMENTED** [review]
Some comments regarding imports as well.

File: src/utils/constants.ts (hunk 4 above)
Comment (an7e) [reply src/utils/constants.ts:2]: added accidentally

File: src/components/httpRoutes/validateCommands.ts [hunk 5]
```diff
@@ -92,6 +92,45 @@ export function validateCommandAPIParameters(requestBody: any): ValidateParams {
...
       }
+    } else if (command === PROTOCOL_COMMANDS.ENCRYPT) {
+      if (!requestBody.blob) {
+        return {
+          valid: false,
+          reason: 'Missing required parameter: "blob"',
+          status: 400
+        }
+      }
+      if (!requestBody.encoding) {
```
Comment (alexcos20) [inline src/components/httpRoutes/validateCommands.ts:103]: can we make encoding default to "string" ?

File: src/components/httpRoutes/validateCommands.ts [hunk 6]
```diff
@@ -92,6 +92,45 @@ export function validateCommandAPIParameters(requestBody: any): ValidateParams {
...
+        }
+      }
+      if (!['String', 'Base58'].includes(requestBody.encoding)) {
+        return {
+          valid: false,
+          reason: 'Invalid parameter: "encoding" must be String | Base58',
+          status: 400
+        }
+      }
+      if (!requestBody.encryptionType) {
```
Comment (alexcos20) [inline src/components/httpRoutes/validateCommands.ts:117]: can we make encryptionType default to "ECIES" please?

File: src/components/httpRoutes/validateCommands.ts (hunk 5 above)
Comment (an7e) [reply src/components/httpRoutes/validateCommands.ts:103]: no problem

File: src/components/httpRoutes/validateCommands.ts (hunk 6 above)
Comment (an7e) [reply src/components/httpRoutes/validateCommands.ts:117]: no problem

**Review by alexcos20 — APPROVED** [review]

**Pull request merged**